tubemq-client/src/main/java/org/apache/tubemq/client/producer/SimpleMessageProducer.java

```diff
@@ -335,11 +336,13 @@ private void checkMessageAndStatus(final Message message) throws TubeClientExcep
     }
 
     private byte[] encodePayload(final Message message) {
```

Comment:
It would be better to put the message compression operation in the Message class. It is open to the user and can avoid the situation where the Message data has been compressed when it is sent and then compressed again.